File: common/src/main/java/com/kt/kol/common/model/ResponseStdVO.java
package com.kt.kol.common.model;

import java.time.LocalDateTime;

public record ResponseStdVO<T>(
		T data,
		LocalDateTime timestamp,
		String message,
		boolean success) {

	public static <T> ResponseStdVO<T> success(T data) {
		return new ResponseStdVO<T>(
				data,
				LocalDateTime.now(),
				"처리성공",
				true);
	}

	public static ResponseStdVO<Void> businessError() {
		return new ResponseStdVO<Void>(
				null,
				LocalDateTime.now(),
				"처리 오류가 발생했습니다",
				false);
	}

	public static ResponseStdVO<Void> businessError(String message) {
		return new ResponseStdVO<Void>(
				null,
				LocalDateTime.now(),
				message,
				false);
	}

	public static ResponseStdVO<Void> systemError() {
		return new ResponseStdVO<Void>(
				null,
				LocalDateTime.now(),
				"시스템 오류가 발생했습니다",
				false);
	}

	public static <T> ResponseStdVO<T> validationError(T validationDetails) {
		return new ResponseStdVO<T>(
				null,
				LocalDateTime.now(),
				"입력 값 검증 실패",
				false);
	}

	public static ResponseStdVO<Void> notFound(String message) {
		return new ResponseStdVO<Void>(
				null,
				LocalDateTime.now(),
				message,
				false);
	}
}

File: common/src/main/kotlin/com/kt/kol/common/model/ResponseStdVO.kt
package com.kt.kol.common.model

import java.time.LocalDateTime

data class ResponseStdVO<T>(
    val data: T?,
    val timestamp: LocalDateTime,
    val message: String,
    val success: Boolean
) {
    companion object {
        fun <T> success(data: T): ResponseStdVO<T> = ResponseStdVO(
            data = data,
            timestamp = LocalDateTime.now(),
            message = "처리성공",
            success = true
        )

        fun businessError(): ResponseStdVO<Void?> = ResponseStdVO(
            data = null,
            timestamp = LocalDateTime.now(),
            message = "처리 오류가 발생했습니다",
            success = false
        )

        fun businessError(message: String): ResponseStdVO<Void?> = ResponseStdVO(
            data = null,
            timestamp = LocalDateTime.now(),
            message = message,
            success = false
        )

        fun systemError(): ResponseStdVO<Void?> = ResponseStdVO(
            data = null,
            timestamp = LocalDateTime.now(),
            message = "시스템 오류가 발생했습니다",
            success = false
        )

        fun <T> validationError(validationDetails: T? = null): ResponseStdVO<T> = ResponseStdVO(
            data = validationDetails,
            timestamp = LocalDateTime.now(),
            message = "입력 값 검증 실패",
            success = false
        )

        fun notFound(message: String): ResponseStdVO<Void?> = ResponseStdVO(
            data = null,
            timestamp = LocalDateTime.now(),
            message = message,
            success = false
        )

        fun <T> error(message: String): ResponseStdVO<T> = ResponseStdVO(
            data = null,
            timestamp = LocalDateTime.now(),
            message = message,
            success = false
        )
    }
}